design/cryptography.md

```diff
@@ -0,0 +1,127 @@
+# Obscuro Cryptography
+
+
+Privacy in Obscuro is achieved by:
+1. using local databases that run inside TEEs and write only encrypted data to disk
+2. the ledger (list of user transactions) is stored encrypted as rollups in Ethereum
+3. users submitting transactions encrypted with a well known key (the Obscuro Key)
+
+
+The first element can be achieved by using existing solutions like [edgelessDB](https://www.edgeless.systems/products/edgelessdb/).
+
+To minimise the complexity for the second and third, we propose to start with a securely generated piece of entropy called: "The Master Seed"(MS),
+and then use that to deterministically derive the keys necessary to achieve the other goals.
+
+An additional complexity is that Obscuro has the requirement to support temporary privacy, and reveal transactions after a period of time has elapsed.
+
+
+## Scope
+
+1. Generating the Master seed
+2. Derive the Obscuro Key
+3. Encrypting transactions such that they can be revealed
+4. Reveal the rollup transactions
+
+## 1. Generating the Master seed
+- Use the random number generation available inside the TEEs
+- Generation happens when the central sequencer bootstraps the network
+
+
+## 2. Derive the Obscuro Key
+- The Obscuro Key is a Key-Pair derived from the Master Seed + Genesis Rollup
+- The Public key will be published to the Management contract and used by all obscuro tooling ( like the wallet extension ) to encrypt transactions
+- The Private key will be determined by other enclaves by deriving the Master Seed + Genesis Rollup 
+  - The Master Seed will be shared using the attestation process
```

Comment:
Reworded as where the validators get the Master Seed and Genesis rollup from so that they can derive the private key.

Actually, why do Validators need to have the Private key ?
Added that as question at the end.